google/cloud/storage/internal/curl_wrappers_disable_sigpipe_handler_test.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/curl_wrappers.h"
+#include "google/cloud/storage/oauth2/google_credentials.h"
+#include <gmock/gmock.h>
+#include <csignal>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+namespace {
+
+extern "C" void test_handler(int) {}
+
+/// @test Verify that installing the libraries
+TEST(CurlWrappers, SigpipeHandlerDisabledTest) {
+#if !defined(SIGPIPE)
+  return;  // nothing to do
+#elif LIBCURL_VERSION_NUM > 0x072900
+  // libcurl <= 7.29.0 installs its own signal handler for SIGPIPE during
+  // curl_global_init(). Unfortunately 7.29.0 is the default on CentOS-7, and
+  // the tests here fails. We simply skip the test with this ancient library.
+  auto initial_handler = std::signal(SIGPIPE, &test_handler);
+  CurlInitializeOnce(ClientOptions(oauth2::CreateAnonymousCredentials())
+                         .set_enable_sigpipe_handler(false));
+  auto actual = std::signal(SIGPIPE, initial_handler);
+  EXPECT_NE(actual, SIG_IGN);
```

Comment:
This expectations looks superfluous given the next one.

Reply:
Removed.